### PR TITLE
WIP upgrade to script setup

### PR DESCRIPTION
**Pull request by JonasWijne**

### CODE REVIEW

File: src/components/AvatarImg.vue [hunk 1]
```diff
@@ -48,45 +48,35 @@
...
 
-            return {
-                src,
-                status,
-            };
-        },
-    };
+    const src = computed(() => {
+        if (!status.value || !status.value.avatar) {
+            return `https://avatars.dicebear.com/4.5/api/jdenticon/${encodeURI(<string>props.id)}.svg?m=14&b=%23ffffff`;
```
Comment (JonasWijne) [inline src/components/AvatarImg.vue:78]: no change

File: src/components/ChatInput.vue [hunk 2]
```diff
@@ -119,299 +119,269 @@
...
+
+    const getActionMessage = computed(() => {
+        switch (action.value.message.type) {
+            case MessageTypes.QUOTE:
+                return (action.value.message.body as QuoteBodyType).message;
+            case MessageTypes.STRING:
+                return action.value.message.body;
+            case MessageTypes.FILE:
+                if (action.value.message.body.type === FileTypes.RECORDING) return 'Voice message';
+                return action.value.message.type;
```
Comment (JonasWijne) [inline src/components/ChatInput.vue:378]: wasnt present and should be handled this way

File: src/components/ChatList.vue [hunk 3]
```diff
@@ -34,13 +33,17 @@
...
                         "
+                        @click="showAddUserDialog = true"
                     >
                         <i class="fas fa-plus"></i>
                     </button>
                     <h1 class="collapsed-bar:hidden pt-1">Messages</h1>
                 </div>
                 <div class="ml-auto collapsed-bar:m-0 collapsed-bar:mb-2 hidden md:block relative">
                     <button
+                        :class="{
```
Comment (JonasWijne) [inline src/components/ChatList.vue:44]: https://github.com/threefoldtech/twin_aydo/pull/156/files?diff=split&w=1#r725866043

File: src/components/ChatList.vue [hunk 4]
```diff
@@ -89,10 +89,9 @@
             </div>
             <div v-if="!collapsed" class="mt-1 mx-2 relative rounded-md shadow-sm">
                 <div class="absolute inset-y-0 left-0 pl-3 flex items-center pointer-events-none">
-                    <SearchIcon class="h-5 w-5 text-gray-400" aria-hidden="true" />
+                    <SearchIcon aria-hidden="true" class="h-5 w-5 text-gray-400" />
```
Comment (JonasWijne) [inline src/components/ChatList.vue:92]: https://github.com/threefoldtech/twin_aydo/pull/156/files?diff=split&w=1#r725866043

File: src/components/ChatList.vue [hunk 5]
```diff
@@ -104,6 +103,7 @@
                         rounded-md
                     "
                     placeholder="Search"
+                    type="text"
```
Comment (JonasWijne) [inline src/components/ChatList.vue:106]: https://github.com/threefoldtech/twin_aydo/pull/156/files?diff=split&w=1#r725866043

File: src/components/ChatList.vue [hunk 6]
```diff
@@ -122,15 +122,17 @@
                     class="text-center collapsed-bar:hidden mt-4"
                 >
                     <p>No messages yet</p>
-                    <button @click="sendUpdate(true)" class="mt-2 border rounded-full px-4">Add a contact</button>
+                    <button class="mt-2 border rounded-full px-4" @click="sendUpdate(true)">Add a contact</button>
                 </div>
                 <div
-                    class="flex flex-col justify-center items-center pt-2 px-2 collapsed-bar:px-0"
                     v-if="filteredChats && filteredChats.length"
+                    class="flex flex-col justify-center items-center pt-2 px-2 collapsed-bar:px-0"
```
Comment (JonasWijne) [inline src/components/ChatList.vue:129]: https://github.com/threefoldtech/twin_aydo/pull/156/files?diff=split&w=1#r725866043

File: src/components/ContactAdd.vue [hunk 7]
```diff
@@ -4,12 +4,12 @@
             <a
                 v-for="item in navigation"
                 :key="item.name"
-                class="nav-link grid-cols-6 text-center py-2 rounded-xl font-normal"
                 :class="[
                     activeItem === item.name
                         ? 'bg-primary text-white'
                         : 'bg-gray-100 text-black hover:bg-gray-200 transition duration-300',
                 ]"
+                class="nav-link grid-cols-6 text-center py-2 rounded-xl font-normal"
```
Comment (JonasWijne) [inline src/components/ContactAdd.vue:12]: https://github.com/threefoldtech/twin_aydo/pull/156/files?diff=split&w=1#r725866043

File: src/components/MessageCard.vue [hunk 8]
```diff
@@ -38,7 +45,7 @@
                         'rounded-bl-xl': isLastMessage,
                     }"
                 >
-                    <header class="p-4 pt-2 pb-2 font-bold my-message:bg-accent-300" v-if="isFirstMessage && isGroup">
+                    <header class="p-4 pt-2 pb-2 font-bold" v-if="isFirstMessage && isGroup">
```
Comment (JonasWijne) [inline src/components/MessageCard.vue:48]: was ugly

File: src/components/MessageContentType/FileShareContent.vue [hunk 9]
```diff
@@ -1,17 +1,18 @@
 <template>
     <a
+        v-if="message.body"
         class="px-4 my-2 my-message:bg-accent-200 cursor-pointer"
         @click="visitFileInMessage(message)"
-        v-if="message.body"
     >
-        <h3 class="my-message:text-icon text-center">
+        <p class="my-message:text-icon text-center">
```
Comment (JonasWijne) [inline src/components/MessageContentType/FileShareContent.vue:7]: too big

File: src/components/MessageContentType/FileShareContent.vue [hunk 10]
```diff
@@ -28,14 +29,14 @@
                 <i class="fas fa-share-alt"></i>
             </div>
             <div class="py-1 flex flex-col">
-                <div class="my-message:text-icon">{{ message.body.name }}</div>
+                <div class="my-message:text-icon flex-shrink-0">{{ message.body.name }}</div>
```
Comment (JonasWijne) [inline src/components/MessageContentType/FileShareContent.vue:32]: text shouldn't shrink

File: src/components/MessageContentType/SystemContent.vue [hunk 11]
```diff
@@ -12,9 +12,9 @@
         <button
             class="
                 text-xs text
-                bg-accent-500
+                bg-accent-800
```
Comment (JonasWijne) [inline src/components/MessageContentType/SystemContent.vue:15]: consistency

File: src/components/fileBrowser/Breadcrumbs.vue [hunk 12]
```diff
@@ -20,8 +20,9 @@
                         <i class="fas fa-chevron-right"></i>
                     </span>
                     <span
-                        class="cursor-pointer text-base p-2 rounded-md"
                         v-if="item || i === 0"
+                        :title="item"
```
Comment (JonasWijne) [inline src/components/fileBrowser/Breadcrumbs.vue:24]: hover over shows name

File: src/components/fileBrowser/MainActionButtons.vue [hunk 13]
```diff
@@ -83,128 +83,98 @@
...
-            };
-        },
+    const showCreateFolderDialog = ref(false);
+    const showCreateFileDialog = ref(false);
+    const newFolderInput = ref<HTMLInputElement>();
+    const newFileInput = ref<any>(undefined);
+    const selectedFiles = ref<File[]>([]);
+    const newFileInputArray = ref<File[]>([]);
+    const createFolderErrors = ref<string[]>([]);
+    const addFolderName = ref<string>('');
```
Comment (JonasWijne) [inline src/components/fileBrowser/MainActionButtons.vue:101]: better name

File: src/plugins/Router.ts [hunk 14]
```diff
@@ -8,7 +8,7 @@ import Browser from '@/views/app/Browser.vue';
 import Kutana from '@/views/app/Kutana.vue';
 import Basic from '@/layout/Basic.vue';
 import Chat from '@/views/app/Chat.vue';
-import Single from '@/views/app/Single.vue';
+import Conversation from '@/views/app/Conversation.vue';
```
Comment (JonasWijne) [inline src/plugins/Router.ts:11]: important rename

**Comment by JonasWijne** [comment]
not all files are changed but is needed for all devs because else there will be serious merge conflicts

File: src/components/notifications/InfoNotification.vue [hunk 15]
```diff
@@ -19,12 +19,10 @@
...
     import { Status, Notification } from '@/types/notifications';
 
-    defineProps({
-        notification: {
-            type: Object,
-            required: true,
-        },
-    });
+    interface IProp {
+        notification: Object;
```
Comment (bertmeeuws) [inline src/components/notifications/InfoNotification.vue:23]: Use ts type

File: src/components/ChatList.vue [hunk 16]
```diff
@@ -183,95 +183,57 @@
...
-                showAddUserDialog,
-            };
-        },
+        //@ts-ignore
+        return uniqBy(filteredChats, c => c.chatId);
     });
+
+    const sendUpdate = newVal => {
+        console.log('update it');
+        console.log('test');
```
[the remark this answers is not in the document]
Comment (JonasWijne) [reply src/components/ChatList.vue:234]:
```suggestion
```

**Pull request merged**